lib/hypervisor/hv_kvm/monitor.py

```diff
@@ -61,18 +61,15 @@ class QmpCommandNotSupported(errors.HypervisorError):
   """
   pass
 
+class QmpTimeoutError(errors.HypervisorError):
+  """QMP socket timeout error """
 
-class QmpMessage(object):
+
+class QmpMessage:
   """QEMU Messaging Protocol (QMP) message.
 
   """
-  def __init__(self, data):
-    """Creates a new QMP message based on the passed data.
-
-    """
-    if not isinstance(data, dict):
```

Comment:
In my opinion, the check is unnecessary because:
1. the typehint tell you what is expected
2. such checks are usually useful for user input, which is not the case here. Only the programmer can pass the wrong type.
3. if it makes sense at this point, why hasn't it been done in the many other places?
4. even if performance is not an argument in Python, but if is a jump command that should be avoided if possible

If such a check should remain, then an assert would make sense, which is only active in debug mode.

Reply:
ACK